lib/paper_trail/model_config.rb

```diff
@@ -110,6 +110,14 @@ def version_class
       @_version_class ||= @model_class.version_class_name.constantize
     end
 
+    def versions_association_item_type
+      if @model_class.descends_from_active_record?
+        @model_class.base_class.name
+      else
+        @model_class.name
+      end
+    end
```

Comment:
After playing around for a bit it ended up seeming that we need a bit more checks to get total consistency with existing tests.  The most intricate ended up [in destroy.rb](https://github.com/seanlinsley/paper_trail/blob/7489c3f73b4bd08b357078a144a518ef85e86e27/lib/paper_trail/events/destroy.rb#L19-L23).

Reply:
I don't believe that'll work for the company / management scenario I described here: https://github.com/paper-trail-gem/paper_trail/pull/1108#issuecomment-412995507

Reply:
Oh -- I thought you had already written the tests you were after.  Is this perhaps the goal?  It passes BTW:

```
require "spec_helper"

class Management < Customer
end

RSpec.describe Management, type: :model, versioning: true do
  it { is_expected.to be_versioned }

  it "utilises the base_class for STI classes having no type column" do
    # Customer (and thus Management) have no stored `type` column.
    expect(Management.inheritance_column).to eq("type")
    expect(Management.columns.map(&:name)).not_to include("type")

    customer1 = Customer.create(name: "Cust 1")
    customer2 = Management.create(name: "Cust 2")

    # When creating a Management the version refers to Customer:
    cust_pts = PaperTrail::Version.where(item_type: "Customer")
    expect(cust_pts.count).to eq(2)

    # Nothing gets tracked as being Managment:
    mgmt_pts = PaperTrail::Version.where(item_type: "Management")
    expect(mgmt_pts.count).to eq(0)
  end
end
```

Perhaps I'll throw it in the mix for good measure because to me it seems to addresses what sounded like your suggestion.  But if this isn't the goal then please write a failing test that addresses Company / Management.

Reply:
It looks like that test covers the scenario I described; thanks for writing it. Are you sure that it passes when checking `descendants.empty?`? It'd seem like it'd only pass when we rely on `descends_from_active_record?`.

Reply:
> Are you sure that it passes when checking `descendants.empty?`?

It works fine with `descendants.empty?`, but I've investigated the other possible override on `creation_attributes` instead now.  It ends up a bit more involved since this more generic class method must check to see if it's dealing with a PaperTrail::Version.  (The other override was actually directly creating a method on Reflection objects.)  Pick whichever one you prefer as both approaches work fine.

Reply:
Ah, interesting. I didn't realize that a monkeypatch on `creation_attributes` is effectively global, while the `collection?` monkeypatch is tied specifically to the versions assocation.

Reply:
I'll go back to the `.collection?()` override then, which is specific to a `Reflection` object.